langlib/lang.regexp/src/main/java/org/ballerinalang/langlib/regexp/RegexUtil.java

```diff
@@ -45,8 +46,12 @@ public class RegexUtil {
     static final BArrayType GROUPS_AS_SPAN_ARRAY_TYPE = new BArrayType(SPAN_AS_TUPLE_TYPE);
 
     static final BArrayType GROUPS_ARRAY_TYPE = new BArrayType(GROUPS_AS_SPAN_ARRAY_TYPE);
-    static Matcher getMatcher(BRegexpValue regexpVal, BString inputStr) {
-        return getMatcher(regexpVal, inputStr.getValue());
+    static Matcher getMatcher(BRegexpValue regexpVal, BString inputStr) throws Exception {
+        try {
+            return getMatcher(regexpVal, inputStr.getValue());
+        } catch (Exception e) {
+            throw new PatternSyntaxException("Invalid regex syntax", regexpVal.toString(), -1);
```

Comment:
Why do you have to wrap every exception inside a PatternSyntaxException?

Reply:
The `PatternSyntaxException` thrown in the method signature is handled here.
https://github.com/ballerina-platform/ballerina-lang/pull/39444/files#diff-5b118eb7fd43169d368376a7e901ef3c2a4d6a12c048d784d1c73c3b14beaf53R49

Reply:
Updated in https://github.com/ballerina-platform/ballerina-lang/pull/39444/commits/5456a304cee5572dfe0375a7564ce75c6a4d8b11.